docs/book/src/getting-started/testdata/project/Makefile

```diff
@@ -152,10 +152,10 @@ $(LOCALBIN):
 
 ## Tool Binaries
 KUBECTL ?= kubectl
-KUSTOMIZE ?= $(LOCALBIN)/kustomize-$(KUSTOMIZE_VERSION)
-CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen-$(CONTROLLER_TOOLS_VERSION)
-ENVTEST ?= $(LOCALBIN)/setup-envtest-$(ENVTEST_VERSION)
-GOLANGCI_LINT = $(LOCALBIN)/golangci-lint-$(GOLANGCI_LINT_VERSION)
```

Comment:
How will ensure that the specific version is the version used?
Note that under the bin we can have many versions.
How we would keep the same behaviour with your proposal?

Reply:
Whenever a _new_ version is downloaded the e.g. `$(LOCALBIN)/kustomize` symlink will be changed to point to `$(LOCALBIN)/kustomize-$(KUSTOMIZE_VERSION)` instead. This doesn't allow to do downgrades at the moment though. Initially I had it always recreate the symlink, which would also allow to do downgrades.
Alternatively we could delete the old version as described in my other comment, which would also trigger a symlink update on every change.

Reply:
So, how can this be helpful for a developer? How can this be better than explicitly setting the version and checking things out? If something breaks, you can simply revert and validate. This way, it is clear what version is being used.

Reply:
Maybe there is a slight misunderstanding here: I like the explicit versioning very much, and its much improved as the current version will actually update when you change the version without requiring to remove the old binaries first.
This change does not remove the explicit versioning, and the versioned binaries will still exists in the bin folder. There will just be an additional symlink to the version that was installed last (the symlink is only created during installation, which could be changed to support downgrades to already installed versions).
![image](https://github.com/user-attachments/assets/fff1a95f-dc2f-402b-b320-8ea2147d0628)

It is helpful if you write scripts that use the binaries downloaded by the makefile. You can just use e.g. `bin/kustomize` and there is no need to adapt it to the version currently set in the makefile, and also no need to execute the script with the makefile to pass in the binary to use.
The only relevant one probably is kustomize, but it should be the same behaviour for all of them.